api/python/provisioner/commands/swupgrade.py

```diff
@@ -0,0 +1,147 @@
+        #   - what about apt and other non-yum pkd managers
+        #   (downgrade is another more generic option but it requires
+        #    exploration of depednecies that are updated)
+        # TODO IMPROVE minions might be stopped here in case of rollback,
+        #      options: set up temp ssh config and rollback yum + minion config
+        #      via ssh as a fallback
+        rollback_ctx = None
+        try:
+            ensure_cluster_is_healthy()  # TODO: checker.run do that check too
+
+            checker = Check()
+            try:
+                check_res = checker.run(GroupChecks.SWUPDATE_CHECKS.value)
+            except Exception as e:
+                logger.warning("During pre-flight checks error happened: "
+                               f"{str(e)}")
+            else:
+                decision_maker = SWUpdateDecisionMaker()
+                decision_maker.make_decision(check_result=check_res)
+
+            self._ensure_upgrade_repos_configuration()
+
+            try:
+                _update_component('provisioner', targets)
+
+                # re-apply provisioner configuration to ensure
+                # that updated pillar is taken into account
+                _apply_provisioner_config(targets)
+
+                config_salt_master()
+
+                minion_conf_changes = config_salt_minions()
+
+                for component in COMPONENTS_FOR_UPGRADE:
+                    _update_component(component, local_minion_id())
+            except Exception as exc:
+                raise SWStackUpdateError(exc) from exc
+
+            try:
+                ensure_cluster_is_healthy()
+            except Exception as exc:
+                raise ClusterNotHealthyError(exc) from exc
+
+            # NOTE that should be the very final step of the logic
+            #      since salt client will be restarted so the current
+            #      process might start to wait itself
+            if minion_conf_changes:
+                # TODO: Improve salt minion restart logic
+                # please refer to task EOS-14114.
+                try:
+                    _restart_salt_minions()
+                except Exception:
+                    logger.exception('failed to restart salt minions')
+
+        except Exception as update_exc:
+            # TODO TEST
+            logger.exception('SW Upgrade failed')
+
+            rollback_error = (
+                None if rollback_ctx is None else rollback_ctx.rollback_error
+            )
+            final_error_t = SWUpdateError
+
+            if rollback_error:
+                # unrecoverable state: SW stack is in intermediate state,
+                # no sense to start the cluster ??? verify TODO IMPROVE
+                logger.error(
+                    'Rollback failed: {}'.format(rollback_ctx.rollback_error))
+
+                final_error_t = SWUpdateFatalError
+            elif rollback_ctx is not None:
+                # yum packages are in initial state here
```

Comment:
[that](https://github.com/Seagate/cortx-prvsnr/blob/7bf919462b747634d54d4f5c6978f94b1ec40f0f/api/python/provisioner/commands/__init__.py#L623-L680) errors processing logic should be kept generally since it's important for rollback. We need just to remove any binding with cluster management.

Reply:
Restore the previous logic